pkg/permit/permit.go

```diff
@@ -47,15 +47,36 @@ func (c *Client) AllTenantsCheck(user enforcement.User, action enforcement.Actio
 	return c.enforcement.AllTenantsCheck(user, action, resource)
 }
 
-func (c *Client) GetUserPermissions(user enforcement.User, tenants ...string) (enforcement.UserPermissions, error) {
-	return c.enforcement.GetUserPermissions(user, tenants...)
+func (c *Client) GetUserPermissions(user enforcement.User, opts ...interface{}) (enforcement.UserPermissions, error) {
+	// First, handle the case where a string is passed directly (for backward compatibility)
+	if len(opts) == 1 {
+		if tenant, ok := opts[0].(string); ok {
+			return c.enforcement.GetUserPermissions(user, enforcement.WithTenants([]string{tenant}))
+		}
+	}
+	
+	// Convert generic options to UserPermissionOption
+	permissionOpts := make([]enforcement.UserPermissionOption, 0, len(opts))
+	for _, opt := range opts {
+		if permOpt, ok := opt.(enforcement.UserPermissionOption); ok {
+			permissionOpts = append(permissionOpts, permOpt)
+		}
+	}
+	
+	return c.enforcement.GetUserPermissions(user, permissionOpts...)
+}
+
+// For backward compatibility
+func (c *Client) GetUserPermissionsWithTenants(user enforcement.User, tenants ...string) (enforcement.UserPermissions, error) {
+	return c.enforcement.GetUserPermissions(user, enforcement.WithTenants(tenants))
 }
 
 type PermitInterface interface {
 	Check(user enforcement.User, action enforcement.Action, resource enforcement.Resource) (bool, error)
 	BulkCheck(requests ...enforcement.CheckRequest) ([]bool, error)
 	FilterObjects(user enforcement.User, action enforcement.Action, context map[string]string, resources ...enforcement.ResourceI) ([]enforcement.ResourceI, error)
 	AllTenantsCheck(user enforcement.User, action enforcement.Action, resource enforcement.Resource) ([]enforcement.TenantDetails, error)
-	GetUserPermissions(user enforcement.User, tenants ...string) (enforcement.UserPermissions, error)
+	GetUserPermissions(user enforcement.User, opts ...interface{}) (enforcement.UserPermissions, error)
+	GetUserPermissionsWithTenants(user enforcement.User, tenants ...string) (enforcement.UserPermissions, error)
```

Comment:
Unfortunately like I said earlier that's a breaking change, it should be opposite - the existing function should stay with the same signature, and the new one should include changes.